src/Providers/components/ProviderRow.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { useTranslation } from 'src/internal/i18n';
+import { ProviderResource } from 'src/internal/k8s';
+
+import { getMostSeriousCondition, getStatusType } from '@app/common/helpers';
+import { IStatusCondition } from '@app/queries/types';
+import { StatusIcon } from '@migtools/lib-ui';
+import { ResourceLink } from '@openshift-console/dynamic-plugin-sdk';
+import { Button, Popover } from '@patternfly/react-core';
+import { Td, Tr } from '@patternfly/react-table';
+
+import { NAME, NAMESPACE, READY, TYPE, URL } from './shared';
+import { RowProps } from './TableView';
+interface CellProps {
+  value: string;
+  resource: ProviderResource;
+  t(key: string): string;
+  kind: string;
+}
+const StatusCell = ({ value, resource, t }: CellProps) => {
+  return (
+    <Popover
+      hasAutoWidth
+      bodyContent={
+        <div>
+          {resource?.status?.conditions?.map((condition) => {
+            const severity = getMostSeriousCondition([
+              condition as IStatusCondition,
+            ]);
+            return (
+              <StatusIcon
+                key={condition.message}
+                status={getStatusType(severity)}
+                label={condition.message}
+              />
+            );
+          }) ?? 'No information'}
+        </div>
+      }
+    >
+      <Button variant="link" isInline aria-label={t(value)}>
+        <StatusIcon
+          status={value === 'Yes' ? 'Ok' : 'Error'}
+          label={t(value)}
+        />
+      </Button>
+    </Popover>
+  );
+};
+
+const TextCell = ({ value }: CellProps) => <>{value}</>;
+
+const ProviderLink = ({ value, resource, kind }: CellProps) => (
+  <ResourceLink
+    kind={kind}
+    name={value}
+    namespace={resource?.metadata?.namespace}
+  />
+);
+
+const cellCreator = {
```

Comment:
can we add the `cellCreator` data into the `Field` interface, so a user can add an optional `Render` method ( fall back to `TextCell` ) ?

Reply:
In the simple case one field maps to one cell. However sometimes you need to access the whole entity to render a cell i.e. take a look at status cell which has access to all conditions (although it maps only to Ready condition)
Mapping the whole entity (resource) to the entire visual component (row, card, list entry) allows to handle such cross-field cases.

Reply:
Consider also cross-cell dependencies (i.e. editing or navigation)

Reply:
a cellRenderer is just a component, it can take what ever props we decide (provided all cellRenderes have the same props) so I don't see a problem with it getting the whole resource (of type resource) and maybe some "extraData" (of type object) just to make sure we don't miss something

Reply:
Right. This was not the best example to support this design. 
Let me try to provide some better reasons:
1. multiple fields may map to one cell i.e. expandable cell but you still want to filter based on them
2. they may be cells without a field  i.e.  technical columns like actions, check boxes, drag&drop 
3. cells may not be independent  - they may need to share state for editing/validating content
4. (mentioned already below) keeping the field definition data-only makes it easy to pass (via props) and store(in state or local store)
5. the field may be rendered in a different way depending on the underlying view i.e. card/list/table view

Reply:
nice, so we have `rowFlatData` that has all the data to render on `tableRow` or `card` or something else, the `fieldsMetadata` help any view we will think of to filter, search, display headers.

It makes sense to me to also decouple the `tableColumnsList` from the `fieldsMetadata`, so we will have a separate lists of the columns that we currently show when in table view and the `defualtTableColumns` that we fall back to when no user defined list is available.